### ISSUE TITLE
Can we have  an example of pure C ++ Examples?

### ISSUE THREAD

**Issue by cgisky1980** [saharNooby/rwkv.cpp#112]
Can we have  an example of pure C ++ Examples?
python & pytorch is too big  for the end users.

**Comment by saharNooby** [comment]
>python & pytorch is too big for the end users.

I agree, but pure C++ implementation is blocked by implementing tokenizer, specifically -- Unicode normalization and Unicode regexes in `20B_tokenizer`. Although RWKV World models are easier to support, since their tokenizer does not require Unicode libraries.

I myself have no plans of implementing the tokenizer in C, but I welcome PRs.

**Comment by cgisky1980** [comment]
> > python & pytorch is too big for the end users.
> 
> I agree, but pure C++ implementation is blocked by implementing tokenizer, specifically -- Unicode normalization and Unicode regexes in `20B_tokenizer`. Although RWKV World models are easier to support, since their tokenizer does not require Unicode libraries.
> 
> I myself have no plans of implementing the tokenizer in C, but I welcome PRs.

https://github.com/mlc-ai/tokenizers-cpp

**Comment by saharNooby** [comment]
@cgisky1980 Unfortunately, `tokenizers-cpp` will not work here:
>You also need to turn on c++17 support
